vmm/src/lib.rs

```diff
@@ -743,8 +750,44 @@ impl Vmm {
         socket
             .read_exact(&mut data)
             .map_err(MigratableError::MigrateSocket)?;
-        let config: VmConfig = serde_json::from_slice(&data).map_err(|e| {
-            MigratableError::MigrateReceive(anyhow!("Error deserialising config: {}", e))
+
+        let vm_migration_config: VmMigrationConfig =
+            serde_json::from_slice(&data).map_err(|e| {
+                MigratableError::MigrateReceive(anyhow!("Error deserialising config: {}", e))
+            })?;
+
+        #[cfg(all(feature = "kvm", target_arch = "x86_64"))]
+        // We check the `CPUID` compatibility of between the source vm and destination, which is
+        // mostly about feature compatibility and "topology/sgx" leaves are not relevant.
+        let dest_cpuid = {
+            let vm_config = &vm_migration_config.vm_config.lock().unwrap();
+            #[cfg(feature = "tdx")]
+            let tdx_enabled = vm_config.tdx.is_some();
+            let phys_bits = vm::physical_bits(
+                vm_config.cpus.max_phys_bits,
+                #[cfg(feature = "tdx")]
+                tdx_enabled,
+            );
+            arch::generate_common_cpuid(
```

Comment:
Ok so here we're duplicating the call to `generate_common_cpuid` since we know it's gonna be called later on, but the goal is only to check that we can go forward, correct?
I'm asking because I'm curious how much time this might take, and we might think about reusing `dest_cpuid` instead of losing it at the end of this function.

Reply:
We decide to keep it for now, considering it is a common practice in our current migration code path where we always build structs from scratch and overwrite their states based on stored/migrated value (e.g. device manager, cpuid, etc).